lib/Controller/ConfigController.php

```diff
@@ -138,20 +138,19 @@ public function setConfig(array $values): DataResponse {
 	 * @return DataResponse
 	 */
 	public function setAdminConfig(array $values): DataResponse {
+		$opUrl = $this->config->getAppValue(Application::APP_ID, 'oauth_instance_url', '');
+
 		foreach ($values as $key => $value) {
 			$this->config->setAppValue(Application::APP_ID, $key, trim($value));
 		}
+		if (isset($values['oauth_instance_url']) && $opUrl !== $values['oauth_instance_url']) {
+			$oauthClientInternalId = $this->config->getAppValue(Application::APP_ID, 'nc_oauth_client_id', '');
+			$id = (int) $oauthClientInternalId;
+			$this->oauthService->setClientRedirectUri($id, $values['oauth_instance_url']);
```

Comment:
```suggestion
			$this->oauthService->setClientRedirectUri((int) $oauthClientInternalId, $values['oauth_instance_url']);
```

this would avoid the very generic variable name `$id`